docs/en/week08/08-3.md

```diff
@@ -0,0 +1,266 @@
+$$
+
+To be clear, we show the unit of $\boldsymbol{g}$:
+
+$$
+\boldsymbol{g} = [g_{1}, g_{2}]^\top, \mathbb{R} \rightarrow \mathbb{R}^{2} \\
+\boldsymbol{g}(z) = [w_{1}\cos(z), w_{2}\sin(z)]
+$$
+
+In this case, the the latent variable $z$ is one dimensional. However, if the $z$ has a higher dimension, this could lead to an energy function of $0$  everywhere. Hence, in order to ensure lower energy values for more compatible predictions we introduce a regularized factor. 
+
+
+### Training recap
+
+Given an observation $\boldsymbol{y}$, training the regualarized latent variable model
+
+$$
+E(\boldsymbol{y},z) = C(\boldsymbol{y},\boldsymbol{\tilde{y}}) + R(z)
+$$ 
+
+where $\boldsymbol{\tilde{y}} = \text{Dec}(z)$,
+
+
+- compute 
+
+$$
+F_{\beta}(\boldsymbol{y}) = \underset{z}{\text{softmin}_{\beta}}[E(\boldsymbol{y},z)]
+$$
+
+- Minimise 
+
+$$
+\mathcal{L}(F_{\beta}(\cdot), Y)
+$$
+
+At zero temperature limit, 
+- Compute
+
+$$
+\check{z} = \underset{z}{\arg\min}{E(\boldsymbol{y},z)} \\
+F_{\infty}(\boldsymbol{y}) = \underset{z}{\min}{E(\boldsymbol{y},\check{z})}
+$$
+
+- Minimise 
+
+$$
+\mathcal{L}(F_{\infty}(\cdot), Y)
+$$
+
+
+### Target propagation
+
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/target_prop.png" style="background-color:#DCDCDC;" /><br>
+<b>Figure 1:</b> Target propagation
+</center>
+
+Given an observation $\boldsymbol{y}$, we first compute $\tilde{z}$ which is the initial guess of the latent variable $z$
+
+$$
+\tilde{z} = \text{Enc}(\boldsymbol{y})
+$$
+
+We then initialize $z$ to $\tilde{z}$, here we must ensure that these values are not too far apart from each other. 
+
+$$
+\check{z} = \underset{z}{\arg\min}{E(\boldsymbol{y},z)} \\
+E(\boldsymbol{y},z) = C(\boldsymbol{y}, \boldsymbol{\tilde{y}}) + R(z) + D(z,\tilde{z})
+$$
+
+We now minimise the loss functional in two steps, minimise $\mathcal{L}(F_{\infty}(\cdot), Y)$:
+
+$$
+\theta_{\text{Dec}} = \theta_{\text{Dec}} - \triangledown_{\theta_{\text{Dec}}}C(\boldsymbol{y},\boldsymbol{\tilde{y}})\\ \\
+\theta_{\text{Enc}} = \theta_{\text{Enc}} - \triangledown_{\theta_{\text{Enc}}}D(\check{z},\tilde{z})
+$$
+
+In this way $z$ is constrained in taking only a set of values.
+
+
+## Autoencoder (AE)
+
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/AE.png" style="background-color:#DCDCDC;" /><br>
+<b>Fig. 2</b>: Autoencoder
+</center>
+
+As shown in *Fig. 2*, to summarize at a high level, a very simple form of AE is as follows:
+
+- First, the autoencoder takes in an input and maps it to a hidden state through an affine transformation $\boldsymbol{h} = f(\boldsymbol{W}_h \boldsymbol{x} + \boldsymbol{b}_h)$, where $f$ is an (element-wise) activation function. This is the **encoder** stage. Note that $\boldsymbol{h}$ is also called the **code**.
+- Next, $\hat{\boldsymbol{x}} = g(\boldsymbol{W}_x \boldsymbol{h} + \boldsymbol{b}_x)$, where $g$ is an activation function. This is the **decoder** stage.
```

Comment:
What is `x-hat`? A high-energy conditional variable?
This semester I've changed the content.
Please, do not copy from last year. It's different. And now it's even wrong.

Reply:
fixed